src/overlay/OverlayManagerImpl.cpp

```diff
@@ -332,6 +323,62 @@ OverlayManagerImpl::ledgerClosed(uint32_t lastClosedledgerSeq)
     mFloodGate.clearBelow(lastClosedledgerSeq);
 }
 
+void
+OverlayManagerImpl::decreasePendingCount(Peer& peer)
+{
+    if (peer.getRole() == Peer::WE_CALLED_REMOTE)
+    {
+        assert(mOutboundPendingPeersCount > 0);
+        mOutboundPendingPeersCount--;
+    }
+    else
+    {
+        assert(mInboundPendingPeersCount > 0);
+        mInboundPendingPeersCount--;
+    }
+}
+
+void
+OverlayManagerImpl::increasePendingCount(Peer& peer)
+{
+    if (peer.getRole() == Peer::WE_CALLED_REMOTE)
+    {
+        mOutboundPendingPeersCount++;
+    }
+    else
+    {
+        mInboundPendingPeersCount++;
+    }
+}
+
+void
+OverlayManagerImpl::decreaseAuthenticatedCount(Peer& peer)
+{
+    if (peer.getRole() == Peer::WE_CALLED_REMOTE)
+    {
+        assert(mOutboundAuthenticatedPeersCount > 0);
+        mOutboundAuthenticatedPeersCount--;
+    }
+    else
+    {
+        assert(mInboundAuthenticatedPeersCount > 0);
+        mInboundAuthenticatedPeersCount--;
+    }
+}
+
+void
+OverlayManagerImpl::increaseAuthenticatedCount(Peer& peer)
```

Comment:
I wonder if it would look better to just have a new struct that tracks pending + authenticated, and then have two members using that type, one for inbound and one for outbound? This seems error prone and with the two types mixed, they may influence each other (so potential for DoS of sorts?)